pelican/contents.py

```diff
@@ -470,9 +486,19 @@ def _log_reason(reason):
 def is_valid_content(content, f):
     try:
         content.check_properties()
-        return True
     except NameError as e:
         logger.error(
             "Skipping %s: could not find information about '%s'",
             f, six.text_type(e))
         return False
+
+    result = True
+    if not content.check_save_as():
+        logger.error(
+            "Skipping %s: file %r would be written outside output path",
+            f,
+            content.save_as,
+        )
+        result = False
+
+    return result
```

Comment:
I can do that. I wanted to encourage future additional checks to use the same structure though, which would emit multiple error messages for things which are wrong instead of stopping at the first wrong thing.